### PR TITLE
Annotation binding

### PR DESCRIPTION
**Pull request by victorg1991**
Hi!

This PR contains two function to allow annotation binding, you can do this:

- Bind an annotation array observable
```swift 
observableOfAnnotationArray.bindTo(mapView.rx.annotation)
```

- Bind an annotation observable
```swift 
observableOfAnnotation.bindTo(mapView.rx.annotation)
```

- Bind and observable of objects (for example a array of bus stops with coordinate information)
```swift
observableOfObjects
     .bindTo(mapView.rx.annotation) { busStop in
           let coordinates
                            = CLLocationCoordinate2D(latitude: stop.latitude, longitude: stop.longitude) 
           let annotation = MKPointAnnotation()
           annotation.coordinate = coordinates
           annotation.title = stop.address

           return annotation
     }
}
```

### CODE REVIEW

File: Pod/Classes/MKMapView+Rx.swift [hunk 1]
```diff
@@ -226,4 +226,36 @@ extension Reactive where Base : MKMapView {
...
+                }
+                .bindTo(self.annotations)
+            }
+    }
+    
+    public func annotations<O: ObservableType> (_ source: O)
+        -> Disposable where O.E == [MKAnnotation] {
+        return source.subscribe(AnyObserver { event in
+            if case let .next(element) = event {
+                self.base.addAnnotations(element as! [MKAnnotation])
```
Comment (shams-ahmed) [inline Pod/Classes/MKMapView+Rx.swift]: `MKMapView+Rx.swift:248:50: Forced cast of '[MKAnnotation]' to same type has no effect`

**Review by shams-ahmed — CHANGES_REQUESTED** [review]
Looking good, any chance you can remove the warning?

**Comment by victorg1991** [comment]
@shams-ahmed done! :)

**Comment by shams-ahmed** [comment]
@victorg1991 thanks

**Pull request merged**